src/GitVersionCore/Core/Abstractions/IRepositoryStore.cs

```diff
@@ -40,7 +40,6 @@ public interface IRepositoryMetadataProvider
         IEnumerable<Tuple<ITag, SemanticVersion>> GetValidVersionTags(string tagPrefixRegex, DateTimeOffset? olderThan = null);
 
         bool GetMatchingCommitBranch(ICommit baseVersionSource, IBranch branch, ICommit firstMatchingCommit);
-        string ShortenObjectId(ICommit commit);
         VersionField? DetermineIncrementedField(BaseVersion baseVersion, GitVersionContext context);
 
         int GetNumberOfUncommittedChanges();
```

Comment:
`GetNumberOfUncommittedChanges` could be exposed as `IGitRepository.Commits.UncommitedChanges`, perhaps?

Reply:
I think I will keep it like this in the PR, so that we can have more time to think.